src/test/java/com/wikia/webdriver/testcases/adstests/TestUrlBuilder.java

```diff
@@ -4,277 +4,383 @@
+                            .getUrlForPath((String) data[0], (String) data[10], (String) data[1]),
+                    (String) data[8]);
+            Assertion.assertEquals(new UrlBuilder("staging", (Boolean) data[2], (Boolean) data[3])
+                            .getUrlForPath((String) data[0], (String) data[10], (String) data[1]),
+                    (String) data[9]);
+        }
+    }
+
+    @Test(groups = {"TestUrlBuilder", "unitTests"})
+    public void getWikiaUrlTests() {
+
+        Assertion.assertEquals(new UrlBuilder("preview", false).getWikiGlobalURL(),
+                "http://www.preview.wikia.com");
+        Assertion.assertEquals(new UrlBuilder("sandbox-s1", false).getWikiGlobalURL(),
+                "http://www.sandbox-s1.wikia.com");
+        Assertion.assertEquals(new UrlBuilder("prod", false).getWikiGlobalURL(),
+                "http://www.wikia.com");
+    }
+
+    @Test(groups = {"TestUrlBuilder", "unitTests"})
+    public void loadFirstPageBeforeLangFromAnnotationIsAvailTests() {
 
-  @Test(groups = "TestUrlBuilder")
-  public void urlBuilder() {
-    Configuration.setTestValue("qs", "");
+        Assertion.assertEquals(new UrlBuilder("preview", false).getUrlForWiki(),
+                String.format("http://%s.preview.wikia.com", Configuration.getWikiName()));
+        Assertion.assertEquals(new UrlBuilder("sandbox-s1", false).getUrlForWiki(),
+                String.format("http://%s.sandbox-s1.wikia.com", Configuration.getWikiName()));
+        Assertion.assertEquals(new UrlBuilder("prod", false).getUrlForWiki(),
+                String.format("http://%s.wikia.com", Configuration.getWikiName()));
+        Assertion.assertEquals(new UrlBuilder("preview", true).getUrlForWiki(),
+                String.format("https://%s.preview.wikia.com", Configuration.getWikiName()));
+        Assertion.assertEquals(new UrlBuilder("sandbox-s1", true).getUrlForWiki(),
+                String.format("https://%s.sandbox-s1.wikia.com", Configuration.getWikiName()));
+        Assertion.assertEquals(new UrlBuilder("prod", true).getUrlForWiki(),
+                String.format("https://%s.wikia.com", Configuration.getWikiName()));
+        Assertion.assertEquals(new UrlBuilder("preview", true, true).getUrlForWiki(),
+                String.format("https://%s.preview.wikia.com", Configuration.getWikiName()));
+        Assertion.assertEquals(new UrlBuilder("sandbox-s1", true, true).getUrlForWiki(),
+                String.format("https://%s.sandbox-s1.wikia.com", Configuration.getWikiName()));
+        Assertion.assertEquals(new UrlBuilder("prod", true, true).getUrlForWiki(),
+                String.format("https://%s.wikia.com", Configuration.getWikiName()));
+    }
+
+    @Test(groups = {"TestUrlBuilder", "unitTests"})
+    public void getUrlForPathTests() {
+
+        Assertion.assertEquals(new UrlBuilder("preview", false).getUrlForPath("/wiki/Special:Version"),
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Define a constant instead of duplicating this literal "/wiki/Special:Version" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)